taichi/program/kernel.cpp

```diff
@@ -269,8 +269,9 @@ void Kernel::LaunchContextBuilder::set_arg_ndarray(int arg_id,
                                /*is_device_allocation=*/true);
   TI_ASSERT_INFO(arr.shape.size() <= taichi_max_num_indices,
                  "External array cannot have > {max_num_indices} indices");
-  for (uint64 i = 0; i < arr.shape.size(); ++i) {
-    this->set_extra_arg_int(arg_id, i, arr.shape[i]);
+  for (uint64 i = 0; i < arr.total_shape().size(); ++i) {
+    // FIXME: we should use arr.shape instead of arr.total_shape() here!
```

Comment:
cc: @turbo0628 I believe this is the last step needed for https://github.com/taichi-dev/taichi/issues/5044. ;)